src/errors.ts

```diff
@@ -0,0 +1,68 @@
+
+// Error codes indicating the reson for an error.
+export enum PurchasesErrorCode {
+    UnknownError = 0,
+    PurchaseCancelledError = 1,
+    StoreProblemError = 2,
+    PurchaseNotAllowedError = 3,
+    PurchaseInvalidError = 4,
+    ProductNotAvailableForPurchaseError = 5,
+    ProductAlreadyPurchasedError = 6,
+    ReceiptAlreadyInUseError = 7,
+    InvalidReceiptError = 8,
+    MissingReceiptFileError = 9,
+    NetworkError = 10,
+    InvalidCredentialsError = 11,
+    UnexpectedBackendResponseError = 12,
+    ReceiptInUseByOtherSubscriberError = 13,
+    InvalidAppUserIdError = 14,
+    OperationAlreadyInProgressError = 15,
+    UnknownBackendError = 16,
+    InvalidAppleSubscriptionKeyError = 17,
+    IneligibleError = 18,
+    InsufficientPermissionsError = 19,
+    PaymentPendingError = 20,
+    InvalidSubscriberAttributesError = 21,
+    LogOutAnonymousUserError = 22,
+}
+
+export interface PurchasesError {
+    code: string;
+    message: string;
+    readableErrorCode: string;
+    underlyingErrorMessage: string;
+    userCancelled: boolean | null;
```

Comment:
Is it possible to make a special error only for purchases? So that this boolean is not in all errors

Reply:
it is... but note that this isn't really a new thing, I'm just trying to reflect the current behavior - the field was added dynamically by us just before sending it over: https://github.com/RevenueCat/react-native-purchases/blob/develop/src/index.ts#L803
so my thinking is that if needed, we could add a new type as a separate thing. 

small note that we sort of backed ourselves into a corner with naming for that one, since this is already called `PurchasesError`

Reply:
Yeah naming is horrible, since it's called Purchases for namespacing but it looks like a purchasing error 🤦

Reply:
It also sucks there's no way of specifying the error a promise will be rejected with. Which makes harder to create a specific error for making purchases

Reply:
I think this field isn't really useful anymore, and hard to maintain, since we have to add it manually, and if more error codes map to "userCancelled = true", we need to add them everywhere. 

So I marked the field as deprecated, and added a suggestion to replace with
 
```typescript
code === Purchases.PurchasesErrorCode.PurchaseCancelled
``` 

@vegaro what do you think?